File: metadata/mill-metadata-core/src/main/java/io/qpointz/mill/metadata/domain/core/ConceptTarget.java
package io.qpointz.mill.metadata.domain.core;

import java.util.ArrayList;
import java.util.List;

/**
 * Target reference for a concept - represents a table with optional attributes.
 * Used in ConceptFacet to reference which tables and attributes a concept relates to.
 */
public record ConceptTarget(
    String schema,
    String table,
    List<String> attributes  // optional, defaults to empty list
) {
    /**
     * Constructor with default empty list for attributes.
     */
    public ConceptTarget {
        if (attributes == null) {
            attributes = new ArrayList<>();
        }
    }
    
    /**
     * Get fully qualified table name.
     *
     * @return FQN string (schema.table)
     */
    public String getFqn() {
        return String.format("%s.%s", schema, table);
    }
}

File: metadata/mill-metadata-core/src/main/kotlin/io/qpointz/mill/metadata/domain/core/ConceptTarget.kt
package io.qpointz.mill.metadata.domain.core

/** Describes where a concept applies in the physical model. */
data class ConceptTarget(
    val schema: String? = null,
    val table: String? = null,
    val attributes: List<String> = emptyList()
) {
    /** Fully-qualified table name (`schema.table`). */
    val fqn: String get() = "$schema.$table"
}
